components/sup/src/main.rs

```diff
@@ -30,6 +30,9 @@ extern crate time;
 extern crate tokio_core;
 extern crate url;
 
+#[cfg(test)]
+extern crate tempdir;
```

Comment:
In light of https://github.com/habitat-sh/habitat/issues/5806, we should just use `tempfile` from the beginning.

(Sorry for leading you astray, I just realized it was deprecated yesterday 😢)